pkg/codegen/schema/pulumi.json

```diff
@@ -0,0 +1,504 @@
+                        }
+                    }
+                },
+                { "$ref": "#/$defs/enumTypeSpec" }
+            ]
+        },
+        "objectTypeSpec": {
+            "title": "Object Type Details",
+            "description": "Describes an object type",
+            "type": "object",
+            "properties": {
+                "properties": {
+                    "description": "A map from property name to propertySpec that describes the object's properties.",
+                    "type": "object",
+                    "additionalProperties": {
+                        "$ref": "#/$defs/propertySpec"
+                    }
+                },
+                "required": {
+                    "description": "A list of the names of an object type's required properties. These properties must be set for inputs and will always be set for outputs.",
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                }
+            }
+        },
+        "enumTypeSpec": {
+            "title": "Enum Type Definition",
+            "description": "Describes an enum type",
+            "type": "object",
+            "properties" :{
+                "type": {
+                    "description": "The underlying primitive type of the enum",
+                    "type": "string",
+                    "enum": ["boolean", "integer", "number", "string"]
+                },
+                "enum": {
+                    "description": "The list of possible values for the enum",
+                    "type": "array",
+                    "items": {
+                        "title": "Enum Value Definition",
+                        "type": "object",
+                        "properties": {
+                            "name": {
+                                "description": "If present, overrides the name of the enum value that would usually be derived from the value.",
+                                "type": "string"
+                            },
+                            "description": {
+                                "description": "The description of the enum value, if any",
+                                "type": "string"
+                            },
+                            "value": {
+                                "description": "The enum value itself",
+                                "type": ["boolean", "integer", "number", "string"]
+                            },
+                            "deprecationMessage": {
+                                "description": "Indicates whether or not the value is deprecated.",
+                                "type": "string"
+                            }
+                        },
+                        "required": ["value"]
+                    }
+                }
+            },
+            "required": ["type", "enum"]
+        },
+        "resourceSpec": {
```

Comment:
I'm guessing this is the hand-authored part. I can read it easier! Thanks.

Reply:
That's right.